addons/survey/static/src/js/survey_form.js

```diff
@@ -46,6 +53,34 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend({
     // Handlers
     // -------------------------------------------------------------------------
 
+    _onKeyPress: function (event) {
+        var self = this;
+        var aIndex = 'a'.charCodeAt(0);
+        var zIndex = 'z'.charCodeAt(0);
+        var code = event.keyCode;
+        var insideTextArea = this.$("textarea").is(":focus");
+        var $choiceInputs = this.$("input[type='radio'],input[type='checkbox']");
+        // Handle Start / Next / Submit
+        if (code === 13 && !insideTextArea) {  // Enter : go Next
+            event.preventDefault();
+            this.$("button.btn-primary").click();
+        } else if (!this.options.readonly
```

Comment:
If it's readonly do you have to listen to anything at all?
Could be moved to avoid registering the keypress listener in the start method IMO.